CopyToClipboard component (#1697)
looker-open-source/components: www/src/documentation/components/actions/button-copy-to-clipboard.mdx

---
title: CopyToClipboard
github: Button/ButtonCopyToClipboard.tsx
storybook: true
---

`CopyToClipboard` button copies a string pass as `content` prop to clipboard.

```jsx

> ;() => {
>   const content = 'here is some text to be copied'
>   return <CopyToClipboard content={content} />
> }

Reviewer: ```suggestion
<CopyToClipboard content='here is some text to be copied' />
```

```

> ### Change Label

Reviewer: ```suggestion
### Label
```

`CopyToClipboard` offers props `children` and `success` both can be used to change the labels on the button by passing a string.

```jsx

> ;() => {
>   const content = 'here is some text to be copied'
>   const children = 'Copy Something'
>   const success = 'it was copied'
> 
>   return (
>     <CopyToClipboard content={content} success={success}>
>       {children}
>     </CopyToClipboard>
>   )
> }

Reviewer: ```suggestion
<CopyToClipboard 
  content="here is some text to be copied" 
  success="'it was copied'"
>
  Copy Something'
</CopyToClipboard>
```

```

> ### Add Component

Reviewer: ```suggestion
### Customize Components
```

> ### Add Component
> 
> `CopyToClipboard` offers props `children` and `success` both can be used to change the component displayed.

Reviewer: ```suggestion
### Customize Components

`CopyToClipboard` `children` and `success` support taking a `JSX.Element` instead of a simple string if you prefer.```

```jsx

> ;() => {
>   const content = 'here is some text to be copied'
>   return (
>     <CopyToClipboard content={content} success={<Button>Success</Button>}>
>       <Button>Copy stuff</Button>
>     </CopyToClipboard>
>   )
> }

Reviewer: ```suggestion
<CopyToClipboard 
  content="here is some text to be copied" 
  success={<Button>Success</Button>}
>
  <Button>Copy stuff</Button>
</CopyToClipboard>
```
